File: Python/7py 2/longest_words.py
"""
|--------------------------------------------------------------------------
| Thinkful - List Drills: Longest word
|--------------------------------------------------------------------------
|
"""

def longest(words):
   return max([len(i) for i in words])


def longest(words):
   return max(map(len, words))


def longest(words):
   return len(max(words, key=len))


print(longest(['simple', 'is', 'better', 'than', 'complex']), 7)
print(longest(['explicit', 'is', 'better', 'than', 'implicit']), 8)
print(longest(['beautiful', 'is', 'better', 'than', 'ugly']), 9)

File: PHP/7php 2/longest_words.php
<?php
/*
|--------------------------------------------------------------------------
| Thinkful - List Drills: Longest word
|--------------------------------------------------------------------------
|
*/

function longest($words) {
   return max(array_map(function($i) {return strlen($i);}, $words));
}


function longest(array $words): int {
   return max(array_map('strlen',$words));
}


function longest($words) {
   return array_reduce($words, function($a,$b) {return max($a,strlen($b));} ); 
}


print_r(longest(['simple', 'is', 'better', 'than', 'complex']) . ' ' . 7 . '
');
print_r(longest(['explicit', 'is', 'better', 'than', 'implicit']) . ' ' . 8 . '
');
print_r(longest(['beautiful', 'is', 'better', 'than', 'ugly']) . ' ' . 9 . '
');